src/cli/commands/pack.js

```diff
@@ -65,10 +66,17 @@ export async function packTarball(
     filters = filters.concat(ignoreLinesToRegex(['!/' + main]));
   }
 
-  // include bundledDependencies
-  if (bundledDependencies) {
-    const folder = config.getFolder(pkg);
-    filters = ignoreLinesToRegex(bundledDependencies.map((name): string => `!${folder}/${name}`), '.');
+  // include bundleDependencies
+  let bundleDependenciesFiles = [];
+  if (bundleDependencies) {
+    for (const dependency of bundleDependencies) {
+      const dependencyList = depsFor(dependency, config.cwd);
```

Comment:
What is this `depsFor` used for? Why can't we just walk the `node_modules/${dependency}` tree?

Reply:
`depsFor` simply returns a list of all of the dependencies for a given package so in a sense it is "walking the tree". I agree with your concerns about not adding unnecessary extra packages to the project. Do you know if yarn has a built in util for getting this info? If not I think it is safe to bring this package in.

Reply:
What I don't understand is: wouldn't something like this work without `depsFor` (pseudo code)?

```
const FOLDERS_IGNORE_NO_NODE_MODULES = new Set(FOLDERS_IGNORE);
FOLDERS_IGNORE_NO_NODE_MODULES.delete('node_modules');

for (const dependency of bundleDependencies) {
  const depPath = getPathForDependency(dependency);

  const filesForBundledDep = await fs.walk(depPath, null, new Set(FOLDERS_IGNORE_NO_NODE_MODULES));
  bundleDependenciesFiles = bundleDependenciesFiles.concat(filesForBundledDep);
}
```

Since `walk` would then traverse the whole `node_modules` tree and pack them all.

Reply:
@arcanis The issue is that this pseudo code doesn't actually walk the dependency graph for a dependency and only walks the directory structure. If we have a dependency `a` which has `b` and `c` as dependencies we get this tree structure:

node_modules/a
node_modules/b
node_modules/c

And `const depPath = getPathForDependency(dependency);` would only walk `node_modules/a` and miss the other ones. Thats why I use `depsFor` as I need more than the "top" level bundled dependencies. Let me know if this make sense or if there is something I am missing.

Reply:
I see, that makes sense. I'm not fond of adding a new dep, but I don't see an easy replacement at the moment, so we can start with that.